py/tests/release.py

```diff
@@ -180,33 +174,77 @@ def node_is_online(api):
 def wait_all_nodes_are_online(apis, timeout_seconds):
     wait(lambda: all([node_is_online(api) for api in apis]), timeout_seconds, sleep_seconds=1)
 
+def wait_all_nodes_are_offline(apis, timeout_seconds):
+    wait(lambda: not any([node_is_online(api) for api in apis]), timeout_seconds, sleep_seconds=1)
+
 def executable(temp_dir):
-    return os.path.join(temp_dir, "bin", "epoch")
+    filename = os.path.join(temp_dir, package_basepath(), "bin", "aeternity")
+    if sys.platform == "win32":
+        return filename + ".cmd"
+    return filename
+
+def extract_package(package_name, temp_dir):
+    ext = os.path.splitext(package_name)[1]
+    if ext == ".zip":
+        extract_ziparchive(package_name, temp_dir)
+    elif ext == ".gz":
+        extract_tarball(package_name, temp_dir)
+    else:
+        sys.exit("ERROR: Unsupported package file format " + ext)
 
 def extract_tarball(tarball_name, temp_dir):
     print("Extracting tar to " + temp_dir)
-    os.system("tar xC " + temp_dir + " -f " + tarball_name)
+    subprocess.call("tar -xC " + temp_dir + " -f " + tarball_name, shell=True)
 
-def stop_node(temp_dir):
-    print("Stopping " + temp_dir)
-    os.system(executable(temp_dir) + " stop")
+def extract_ziparchive(zip_name, temp_dir):
+    print("Extracting zip to " + temp_dir)
+    subprocess.call("unzip -q " + zip_name + " -d " + temp_dir, shell=True)
 
-def start_node(temp_dir):
+def find_pid(line, listen_address, acc):
+    parts = [p for p in line.split(' ') if not p == '']
+    if not len(parts) == 5:
+        return acc
+    if not parts[1] == listen_address:
+        return acc
+    return parts[4]
+
+def stop_node(node, temp_dir):
     binary = executable(temp_dir)
-    assert os.path.isfile(binary)
-    assert os.access(binary, os.X_OK)
+    assert os.path.isfile(binary), "Not a file: " + binary
+    assert os.access(binary, os.X_OK), "Can't access file: " + binary
+    print("Stopping " + binary)
+    os.chdir(temp_dir)
+    if sys.platform == "win32":
+        # We need to find the PID listening on the right port
+        listen_address = SETUP[node]["listen_address"]
+        result_lines = subprocess.check_output("netstat -ano -p TCP").decode("utf-8").split("\r\n")
+        pid = functools.reduce(lambda acc, r: find_pid(r, listen_address, acc), result_lines, "")
+        os.kill(int(pid), signal.SIGTERM)
+    else:
+        return subprocess.call(binary + " stop", shell=True)
+
+def start_node(node, temp_dir):
+    binary = executable(temp_dir)
+    assert os.path.isfile(binary), "Not a file: " + binary
+    assert os.access(binary, os.X_OK), "Can't access file: " + binary
     print("Starting " + binary)
     os.chdir(temp_dir)
-    os.system('ERL_FLAGS="-config `pwd`/p.config" bin/aeternity start')
+    erl_flags = " ".join(SETUP[node]["config"])
+    if sys.platform == "win32":
+        sub_env = os.environ
+        sub_env["ERL_FLAGS"] = erl_flags
+        subprocess.Popen([binary, "console"], env=sub_env)
```

Comment:
Is there something closer to how user would pass flags?

Reply:
Using `ERL_FLAGS` is the way any user would do that. The difference to the previous approach is that one can't pass multiple configuration file arguments on Windows, which is why I had to refactor it.